src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.IsFinite.cs

```diff
@@ -0,0 +1,137 @@
+        /// If <paramref name="x"/> is empty, <see langword="false"/> is returned.
+        /// </returns>
+        public static bool IsFiniteAll<T>(ReadOnlySpan<T> x)
+            where T : INumberBase<T> =>
+            !x.IsEmpty &&
+            (AlwaysFinite<T>() || All<T, IsFiniteOperator<T>>(x));
+
+        /// <summary>Computes whether any of the values in the specified tensor is finite.</summary>
+        /// <param name="x">The tensor, represented as a span.</param>
+        /// <returns>
+        /// <see langword="true"/> if any of the values in <paramref name="x"/> is finite; otherwise, <see langword="false"/>.
+        /// If <paramref name="x"/> is empty, <see langword="false"/> is returned.
+        /// </returns>
+        public static bool IsFiniteAny<T>(ReadOnlySpan<T> x)
+            where T : INumberBase<T> =>
+            !x.IsEmpty &&
+            (AlwaysFinite<T>() || Any<T, IsFiniteOperator<T>>(x));
+
+        /// <summary>Gets whether all values of the specified type are always finite.</summary>
+        private static bool AlwaysFinite<T>() =>
+            typeof(T) == typeof(decimal) ||
+            typeof(T) == typeof(sbyte) || typeof(T) == typeof(byte) ||
+            typeof(T) == typeof(short) || typeof(T) == typeof(ushort) || typeof(T) == typeof(char) ||
+            typeof(T) == typeof(int) || typeof(T) == typeof(uint) ||
+            typeof(T) == typeof(long) || typeof(T) == typeof(ulong) ||
+            typeof(T) == typeof(nint) || typeof(T) == typeof(nuint) ||
+            typeof(T) == typeof(Int128) || typeof(T) == typeof(UInt128);
+
+        /// <summary>T.IsFinite(x)</summary>
+        private readonly struct IsFiniteOperator<T> : IBooleanUnaryOperator<T>
+            where T : INumberBase<T>
+        {
+            public static bool Vectorizable => true;
+
+            public static bool Invoke(T x) => T.IsFinite(x);
+
+#if NET10_0_OR_GREATER
+            public static Vector128<T> Invoke(Vector128<T> x) => Vector128.IsFinite(x);
+            public static Vector256<T> Invoke(Vector256<T> x) => Vector256.IsFinite(x);
+            public static Vector512<T> Invoke(Vector512<T> x) => Vector512.IsFinite(x);
+#else
+            public static Vector128<T> Invoke(Vector128<T> x)
+            {
+                if (typeof(T) == typeof(float))
+                {
+                    Vector128<uint> positiveInfinityBits = Vector128.Create(0x7F80_0000u);
+                    return (~Vector128.Equals(~x.AsUInt32() & positiveInfinityBits, Vector128<uint>.Zero)).As<uint, T>();
+                }
+
+                if (typeof(T) == typeof(double))
+                {
+                    Vector128<ulong> positiveInfinityBits = Vector128.Create<ulong>(0x7FF0_0000_0000_0000u);
+                    return (~Vector128.Equals(~x.AsUInt64() & positiveInfinityBits, Vector128<ulong>.Zero)).As<ulong, T>();
+                }
+
+                return Vector128<T>.AllBitsSet;
```

Comment:
When 'extensions' are more complete, we should be able to move these out to internal polyfills, right?

Reply:
I hope so